index.html

```diff
@@ -250,31 +248,56 @@
           ctx.fillRect(pixelXMin, pixelYMax, pixelXMax - pixelXMin, pixelYMin - pixelYMax);
         }
 
-        // OK Zone (Zielony obszar)
-        const okStartZone = Math.max(minOkBudgetPercentage * ignoreThreshold, scales.x.min);
-
-        for (let x = okStartZone; x <= xMax; x += calculateStep(x)) {
-          const xMinPixel = getPixel(scales.x, x);
-          const xMaxPixel = getPixel(scales.x, x + calculateStep(x));
-
-          const yMinValue = x < ignoreThreshold ? ignoreThreshold : Math.min(minOkBudgetPercentage * x, yMax);
-          const yMaxValue = Math.max(Math.min(maxOkBudgetPercentage * (x + calculateStep(x)), yMax), ignoreThreshold);
+        // Ok Zone Fixed to avoid wrong rendering
+        const pointsOK = [
+          {
+            x: getPixel(scales.x, ignoreThreshold / maxOkBudgetPercentage),
+            y: getPixel(scales.y, ignoreThreshold),
+          },
+          {
+            x: getPixel(scales.x, ignoreThreshold),
+            y: getPixel(scales.y, ignoreThreshold),
+          },
+          {
+            x: getPixel(scales.x, ignoreThreshold),
+            y: getPixel(scales.y, ignoreThreshold * minOkBudgetPercentage),
+          },
+          {
+            x: getPixel(scales.x, xMax),
+            y: getPixel(scales.y, xMax * minOkBudgetPercentage)
+          },
+          {
+            x: getPixel(scales.x, xMax),
+            y: getPixel(scales.y, xMax * maxOkBudgetPercentage)
+          },
+          {
+            x: getPixel(scales.x, xMax / maxOkBudgetPercentage),
+            y: getPixel(scales.y, xMax)
+          },
+        ];
+        ctx.save();
 
-          // Omiń obszary poza widocznym zakresem
-          if (yMinValue > yMax || yMaxValue < yMin) continue;
+        // Ustaw maskę odpowiadającą granicom wykresu
+        ctx.beginPath();
+        ctx.rect(left, top, right - left, bottom - top);
+        ctx.clip(); // O
 
-          const yMinPixel = Math.max(top, getPixel(scales.y, yMaxValue));
-          const yMaxPixel = Math.min(bottom, getPixel(scales.y, yMinValue));
+        // Rysowanie poligonu
+        ctx.beginPath();
+        ctx.fillStyle = okColor;
+        ctx.moveTo(pointsOK[0].x, pointsOK[0].y);
 
-          ctx.fillStyle = okColor;
-          ctx.fillRect(xMinPixel, yMinPixel, xMaxPixel - xMinPixel, yMaxPixel - yMinPixel);
-        }
+        pointsOK.forEach(point => {
+          ctx.lineTo(point.x, point.y);
+        });
+        ctx.closePath();
+        ctx.fill();

```

Comment:
This allows to draw simple poligon (6 points) fill it will color, and then cut to fit the chart area